plugins/www/Optimove.js

```diff
@@ -1,44 +1,204 @@
+var exec = require("cordova/exec");
 
+var inAppInboxUpdatedHandler;
 
-var exec = require('cordova/exec');
+var currentConfig = {
+  pushReceivedHandler: pushReceivedHandler, //function that receives one argument, a push message object
+  pushOpenedHandler: pushOpenedHandler, //function that receives one argument, a push message object
+  inAppDeepLinkHandler: inAppDeepLinkHandler, //expect to be a function that receives one argument, a deepLink data object
+};
 
+document.addEventListener("deviceready", init, false);
 
+function init() {  
+  setHandlersCallBackContext().then(success, (errorMessage) => { console.error(errorMessage); });
+}
+
+
+ function setHandlersCallBackContext() {
+   return new Promise((resolve, reject) => {
+     exec(
+       nativeMessageHandler,
+       reject,
+       "OptimoveSDKPlugin",
+       "setHandlersCallBackContext",
+       []
+     );
+   });
+}
+ 
+function checkIfPendingPushExists() {
+  return new Promise((resolve, reject) => {
+    exec(resolve, reject, "OptimoveSDKPlugin", "checkIfPendingPushExists", []);
+  });
+}
+
+function nativeMessageHandler(message) {
+  if (!message || typeof message === "string") {
+    return;
+  }
+  
+  const handlerName = `${message.type}Handler`;
+  alert(handlerName);
```

Comment:
Can we chop alerting from the plugin implementation?

Reply:
yeah sure, sorry I forgot to delete some debug prints